pkg/apis/pipeline/v1beta1/taskrun_conversion_test.go

```diff
@@ -280,6 +280,40 @@ func TestTaskRunConversionFromDeprecated(t *testing.T) {
 				},
 			},
 		},
+	}, {
+		name: "bundle",
+		in: &v1beta1.TaskRun{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "foo",
+				Namespace: "bar",
+			},
+			Spec: v1beta1.TaskRunSpec{
+				TaskRef: &v1beta1.TaskRef{
+					Name:   "test-bundle-name",
+					Bundle: "test-bundle",
+				},
+			},
+		},
+		want: &v1beta1.TaskRun{
```

Comment:
@abayer Could you please help confirm the expected output here in the test? Wondering if we would want to introduce extra Params when converting back to v1beta1.

Reply:
This output is an invalid TaskRef-- you can't specify both resolver and bundle. I think we have 2 options here:

1. Convert all v1 bundle resolverrefs back to oci bundle syntax. This means while v1beta1 is the stored version, if the user creates a v1 CRD, the thing fetching bundles will be the oci bundle code even if the user specified a resolver ref. If the user creates a v1beta1 CRD, the resolver ref syntax will stay as is. I don't love this solution because it will result in different behavior depending on whether the user created a v1beta1 or v1 CRD (while v1beta1 is the stored version). AFAIK the bundle resolver doesn't have any additional features compared to the oci bundle syntax, but it might be more performant.

2. Turn on bundle resolution by default (i.e. remove the feature flag `enable-bundles-resolver`) and don't convert v1 resolver refs back to bundles. I prefer this solution but it might be weird to have the bundles resolver turned on by default while others are not.

@abayer I'd also like to hear your thoughts here

Reply:
`name` and `bundle` should both not be there in the `TaskRef` if `ResolverRef` is populated, yes.

I'd lean towards option 2 - I kinda want to turn on the bundle, hub, cluster, and (current, i.e., anonymous) git resolvers by default in v0.41, as part of promoting remote resolution to beta. That is, I think that we can still reserve the right to make changes to the remote resolution API, but am pretty sure we're locked in on the `pipelineRef`/`taskRef` syntax, so enabling them by default seems fine to me, with the ability to disable them via the feature flags still there.

Reply:
Sounds good. I think we don't need to wait for remote resolution to be in beta and the flag to be removed before merging this PR, since we will be moving remote resolution to beta before swapping the stored version of our crds to v1. So @JeromeJu you can just remove the conversion from resolver -> bundle. Andrew, lmk if any concerns about this

Reply:
Works for me!

Reply:
Thanks @lbernick and @abayer.